spec/vehicle_factory_spec.rb

```diff
@@ -0,0 +1,270 @@
+require "spec_helper"
+
+RSpec.describe VehicleFactory do
+  before(:each) do
+    @factory = VehicleFactory.new
+    @wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    @ny_registrations = DmvDataService.new.ny_registrations
+    @three_mock_wa_ev_regs = [
```

Comment:
Oh I like this! You'll learn some more efficient mocking techniques in Mod 3, however there are still certainly times when I go back to manually mocking just like this. To keep the spec files a little cleaner, you might enjoy looking into the concept of fixture files. Essentially, just a directory inside the spec directory called `fixtures` and it holds all our large amounts of test data.